compile errors when adding 15.0.0 as dependency

Trying to upgrade crates-index-diff on docs.rs I got some compile errors: 

```
error[E0425]: cannot find function `serialize` in crate `hex`
   --> /Users/syphar/.cargo/registry/src/github.com-1ecc6299db9ec823/crates-index-diff-15.0.0/src/types.rs:125:26
    |
125 | #[derive(Default, Clone, serde::Serialize, serde::Deserialize, Eq, PartialEq, Debug)]
    |                          ^^^^^^^^^^^^^^^^ not found in `hex`
    |
    = note: this error originates in the derive macro `serde::Serialize` (in Nightly builds, run with -Z macro-backtrace for more info)
``` 
```
error[E0277]: the trait bound `SmartString<LazyCompact>: Serialize` is not satisfied
    --> /Users/syphar/.cargo/registry/src/github.com-1ecc6299db9ec823/crates-index-diff-15.0.0/src/types.rs:125:26
     |
125  | #[derive(Default, Clone, serde::Serialize, serde::Deserialize, Eq, PartialEq, Debug)]
     |                          ^^^^^^^^^^^^^^^^ the trait `Serialize` is not implemented for `SmartString<LazyCompact>`
126  | pub struct CrateVersion {
127  |     /// The crate name, i.e. `clap`.
     |     -------------------------------- required by a bound introduced by this call
     |
     = help: the following other types implement trait `Serialize`:
               &'a T
               &'a mut T
               ()
               (T0, T1)
               (T0, T1, T2)
               (T0, T1, T2, T3)
               (T0, T1, T2, T3, T4)
               (T0, T1, T2, T3, T4, T5)
             and 131 others
note: required by a bound in `types::_::_serde::ser::SerializeStruct::serialize_field`
    --> /Users/syphar/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.147/src/ser/mod.rs:1899:12
     |
1899 |         T: Serialize;
     |            ^^^^^^^^^ required by this bound in `types::_::_serde::ser::SerializeStruct::serialize_field`
```
and some similar errors. 

### how to reproduce: 
in an empty directory: 
- `cargo init`
- `cargo add crates-index-diff`
- `cargo build`

### related links 
- https://github.com/rust-lang/docs.rs/compare/master...syphar:docs.rs:upgrade-git 
- https://github.com/syphar/docs.rs/actions/runs/3546296822/jobs/5955192211

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -17,4 +17,11 @@ jobs:
       run: make test
     - name: docs
       run: cargo doc
+    - name: usage as dependency
+      run: |
+          cargo init user
+          cd user
+          cargo add --path ..
+          cargo check
+           
 

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -37,8 +37,8 @@ http-reqwest = ["git-repository/blocking-http-transport-reqwest-rust-tls"]
 [dependencies]
 git-repository = { version = "0.29.0", default-features = false, features = ["max-performance-safe", "blocking-network-client"] }
 serde = { version = "1", features = ["std", "derive"] }
-hex = "0.4.3"
-smartstring = "1.0.1"
+hex = { version = "0.4.3", features = ["serde"] }
+smartstring = { version = "1.0.1", features = ["serde"] }
 serde_json = "1"
 bstr = "1.0.1"
 thiserror = "1.0.32"

